lib/ansible/galaxy/role.py

```diff
@@ -79,7 +79,10 @@ class GalaxyRole(object):
     META_REQUIREMENTS = (os.path.join('meta', 'requirements.yml'), os.path.join('meta', 'requirements.yaml'))
     ROLE_DIRS = ('defaults', 'files', 'handlers', 'meta', 'tasks', 'templates', 'vars', 'tests')
 
-    def __init__(self, galaxy, api, name, src=None, version=None, scm=None, path=None):
+    def __init__(self, galaxy, api, name=None, src=None, version=None, scm=None, path=None):
+
+        if name is None:
```

Comment:
Do we expect all callers to extract `name` from `src` or should we also check `src is None` here?

Reply:
`name` is set to `role` or derived from `src` or provided directly. A src isn't required, only a name. I just copied the error, but maybe need to wordsmith that...